src/utils/emoji.ts

```diff
@@ -0,0 +1,21 @@
+import { BaseEmoji, CustomEmoji, EmojiData } from 'emoji-mart'
+
+export const getEmojiIcon = (emoji: EmojiData):string => {
+  if ((emoji as BaseEmoji).native) {
+    return (emoji as BaseEmoji).native
+  } else if ((emoji as CustomEmoji).imageUrl) {
+    // noinspection CheckTagEmptyBody
+    return `<i class="fa ${(emoji as CustomEmoji).name}"></i>`
+  }
+  return ''
+}
+
+export const getEmojiShortCode = (emoji: EmojiData):string => {
+  if ((emoji as BaseEmoji).colons) {
+    return (emoji as BaseEmoji).colons
+  } else if ((emoji as CustomEmoji).imageUrl) {
+    // noinspection CheckTagEmptyBody
+    return `:${(emoji as CustomEmoji).name}:`
+  }
```

Comment:
`.colons` does exist on BaseEmoji as well as on CustomEmoji. You'll probably never end up  in the `else if` branch.
To distinguish between BaseEmoji and CustomEmoji, you have to test for `.native` and `.imageUrl`.

Reply:
It doesn't matter if it exists, only if it is filled.

Reply:
Resolved in c6c72bc5